dagfactory/dagfactory.py

```diff
@@ -304,22 +312,35 @@ def load_yaml_dags(
     :param globals_dict: The globals() from the file used to generate DAGs
     :param dags_folder: Path to the folder you want to get recursively scanned
     :param default_args_config_path: The Folder path where defaults.yml exist.
+    :param config_filepath: A YAML path for DAG config.
+    :param config_dict: The DAG dictionary.
+    :param default_args_config_dict: The dictionary that hold default value.
     :param suffix: file suffix to filter `in` what files to scan for dags
     """
+    # TODO: Support ignoring yml files in load_yaml_dags
+    # https://github.com/astronomer/dag-factory/issues/527
     # chain all file suffixes in a single iterator
     logging.info("Loading DAGs from %s", dags_folder)
     if suffix is None:
         suffix = [".yaml", ".yml"]
     candidate_dag_files = []
-    for suf in suffix:
-        candidate_dag_files = list(chain(candidate_dag_files, Path(dags_folder).rglob(f"*{suf}")))
-    for config_file_path in candidate_dag_files:
-        config_file_abs_path = str(config_file_path.absolute())
-        logging.info("Loading %s", config_file_abs_path)
-        try:
-            factory = DagFactory(config_file_abs_path, default_args_config_path=default_args_config_path)
-            factory.generate_dags(globals_dict)
-        except Exception:  # pylint: disable=broad-except
-            logging.exception("Failed to load dag from %s", config_file_path)
-        else:
-            logging.info("DAG loaded: %s", config_file_path)
+
+    if config_filepath:
+        factory = DagFactory(config_filepath=config_filepath, default_args_config_path=default_args_config_path)
+        factory._generate_dags(globals_dict)
+    elif config_dict:
+        factory = DagFactory(config=config_dict, default_args_config_dict=default_args_config_dict)
+        factory._generate_dags(globals_dict)
+    else:
+        for suf in suffix:
+            candidate_dag_files = list(chain(candidate_dag_files, Path(dags_folder).rglob(f"*{suf}")))
+        for config_file_path in candidate_dag_files:
+            config_file_abs_path = str(config_file_path.absolute())
+            logging.info("Loading %s", config_file_abs_path)
+            try:
+                factory = DagFactory(config_file_abs_path, default_args_config_path=default_args_config_path)
+                factory._generate_dags(globals_dict)
+            except Exception:  # pylint: disable=broad-except
+                logging.exception("Failed to load dag from %s", config_file_path)
+            else:
+                logging.info("DAG loaded: %s", config_file_path)
```

Comment:
@pankajastro I was a bit confused about this part, it feels a very similar logic to what I had implemented on #500 , in particular:
https://github.com/astronomer/dag-factory/blob/2fd9c757d564eed1686f24edf353d166db518d64/dagfactory/dagfactory.py#L43-L74

Why not just reuse what was available in the now internal `DagFactory` class, instantiating it with the necessary arguments?

Reply:
Ah, I see, `config_filepath` means YAML DAG filepath instead of "configuration" (that contains defaults) - thanks, @pankajastro for the clarification!